Add heuristic for Perl6 and Turing

Alhadis opened the pull request:
After running Linguist a few times on Turing repositories in the wild (such as [this one](https://github.com/jbrick/highschool_turing)), I feel it could still benefit from an additional heuristic, even if #3040 greatly improved Linguist's classification of Turing from Perl6.

Turing and Perl6 both use different styles of line-comments (`%` versus `#`, respectively), so the heuristic was easy to add. [Surprisingly](http://i.imgur.com/9Fxm1eT.png), I managed to get Perl6 running to test if `%` was invalid syntax at the start of a line. A glaring error message confirmed that it was indeed invalid syntax:

<img src="https://cloud.githubusercontent.com/assets/2346707/15985281/91dd4d6e-302b-11e6-9728-98d3999a3d09.png" width="432" alt="Figure 1" />

Turing's heuristic also checks for `var {name} : {type}`-like patterns, where `{name}` and `{type}` are sequences of word-characters (`\w`). These lines are unlikely to manifest in Perl 6, which retains Perl 5's method of initialising variables ([`my $a = accum 5;`](https://github.com/perl6/perl6-examples/blob/master/categories/best-of-rosettacode/accumulator-factory.pl#L84)).

pchaigno commented:
Nice pull request!

Were you able to test it on a few sets of Perl6/Turing files?

Alhadis commented:
No, actually. Every time I tried installing a locally-built gem, it wound up using some other version that was oblivious to the heuristic I added. I kept giving me the same incorrect results for the [repository I mentioned](https://github.com/jbrick/highschool_turing) as before I added the heuristic.

arfon commented:
:+1: thanks @Alhadis

Alhadis commented:
Always a pleasure, mate.

Alhadis commented:
Always a pleasure, mate.
